### PR TITLE
Virtual alexa integration#394

### LINKED ISSUE #394: Update bst utter and intend to use VirtualAlexa
The existing BSTAlexa piece should be deprecated (and marked to go away in Version 2.0). 

The command-line utter and intend commands should now use VirtualAlexa. Additionally, we should handle the new interaction model.

By default, the commands should look for a file "speechAssets/InteractionModel.json". There should also be an option "--model, -m" for the user to specify a custom location.

The documentation should be updated to reflect the new way that VirtualAlexa works - in particular, for the utter command it is not longer necessary to pass slot values surrounded by  {}.

### PR DESCRIPTION
**Pull request by jperata**
Closes #394 
- Add Bst-virtual-alexa as a wrapper for bespoken/virtual-alexa
- Change bst-launch, bst-intend and bst-utter to use bst-virtual-alexa wrapper and use interaction model as the default schema model.
- Ensures old functionality isn't broken.

Missing:
- Update documentation to reflect changes.

### CODE REVIEW

File: bin/bst-intend.ts [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 import {Global} from "../lib/core/global";
-import {BSTAlexa} from "../lib/client/bst-alexa";
+import {BSTVirtualAlexa} from "../lib/client/bst-virtual-alexa";
 
 program.version(Global.version());
 
 program
     .usage("[options] <Intent> [SlotName=SlotValue...]")
     .option("-u, --url <alexa-skill-url>", "The URL of the Alexa skill to send to - defaults to current proxied skill")
+    .option("-m, --model <interaction-model-path>", "Path to the interaction model file - defaults to ./speechAssets/InteractionModel.json")
```
Comment (jkelvie) [inline bin/bst-intend.ts]: For the interaction model file, I think we should use the default location for the ASK CLI - this is what Amazon is moving all there examples to. It is: `./models/en-US.json`

See example here:
https://github.com/alexa/skill-sample-nodejs-fact/tree/en-US/models

File: bin/bst-utter.ts [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
 import {Global} from "../lib/core/global";
-import {BSTAlexa} from "../lib/client/bst-alexa";
+import {BSTVirtualAlexa} from "../lib/client/bst-virtual-alexa";
 
 program.version(Global.version());
 
 program
     .usage("[options] <utterance>")
     .option("-u, --url <alexa-skill-url>", "The URL of the Alexa skill to speak to - defaults to current proxied skill")
+    .option("-m, --model <interaction-model-path>", "Path to the interaction model file - defaults to ./speechAssets/InteractionModel.json")
```
Comment (jkelvie) [inline bin/bst-utter.ts]: Same as above with regard to the default location for the interaction model.

File: lib/client/bst-virtual-alexa.ts [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+     * @param sampleUtterancesFile The path to the samples utterances file - defaults to {@link BSTAlexa.DefaultSampleUtterancesLocation}.
+     * @param applicationID The application ID. Just makes one up if none is defined.
+     */
+    public constructor(private skillURL: string,
+                       private interactionModel?: string,
+                       private intentSchemaFile?: string,
+                       private sampleUtterancesFile?: string,
+                       private applicationID?: string) {
+        if ((intentSchemaFile || sampleUtterancesFile) && interactionModel) {
+            console.error("Interaction Model and Intent Schema Files shouldn't be sent together. Chose one of those");
```
Comment (jkelvie) [inline lib/client/bst-virtual-alexa.ts]: "The Interaction Model and Intent Schema Files should not both be specified. It should be one or the other."

**Comment by jperata** [comment]
@jkelvie Maintaining the session will be on a different ticket.

**Pull request merged**